### PR TITLE
fix(core): Introduce jitter when re-queueing message if executor is full

### PR DESCRIPTION
**Pull request by ajordens**
- include more meaningful log output (message, attempts, delay, etc.)

### CODE REVIEW

File: keiko-core/src/main/kotlin/com/netflix/spinnaker/q/QueueProcessor.kt [hunk 1]
```diff
@@ -82,8 +85,17 @@ class QueueProcessor(
             }
           }
         } catch (e: RejectedExecutionException) {
-          log.warn("Executor at capacity, immediately re-queuing message", e)
-          queue.push(message)
+          val requeueDelay = Duration.ofSeconds(2 + random.nextInt(5).toLong())
```
Comment (robfletcher) [inline keiko-core/src/main/kotlin/com/netflix/spinnaker/q/QueueProcessor.kt]: Isn't there a `nextLong`?

Comment (robfletcher) [reply keiko-core/src/main/kotlin/com/netflix/spinnaker/q/QueueProcessor.kt]: No, I just looked and there isn't (at least not one with a parameter). Objection withdrawn.

Comment (ajordens) [reply keiko-core/src/main/kotlin/com/netflix/spinnaker/q/QueueProcessor.kt]: 👍

File: keiko-core/src/main/kotlin/com/netflix/spinnaker/q/QueueProcessor.kt [hunk 2]
```diff
@@ -37,9 +39,12 @@ class QueueProcessor(
   private val activator: Activator,
   private val publisher: EventPublisher,
   private val deadMessageHandler: DeadMessageCallback,
-  private val fillExecutorEachCycle: Boolean = false
+  private val fillExecutorEachCycle: Boolean = false,
+  private val requeueDelay : Duration = Duration.ofSeconds(0),
+  private val requeueMaxJitter : Duration = Duration.ofSeconds(0)
```
Comment (robfletcher) [inline keiko-core/src/main/kotlin/com/netflix/spinnaker/q/QueueProcessor.kt]: `Duration` \o/

**Pull request merged**